utils/tfw_logger.cc

```diff
@@ -333,136 +357,269 @@ void stop_daemon()
+ 
+		// Re-initialize logger after daemonizing
+		setup_logger(config);
+	}
+	else
+	{
+		spdlog::info("Starting in handle (foreground) mode");
+	}
+}
 
-		set_sig_handlers();
-
-		std::ofstream pid_file(pid_file_path);
-		if (!pid_file)
-			throw Except("Failed to open PID file");
-		pid_file << getpid();
-		pid_file.close();
-
-		while ((fd = open(dev_path, O_RDWR)) == -1) {
-			if (stop_flag.load(std::memory_order_acquire))
-				goto end;
-			if (errno != ENOENT)
-				throw Except("Can't open device");
-			std::this_thread::sleep_for(wait_for_dev);
+void 
+create_pid_file(const fs::path &pid_file_path)
+{
+	std::ofstream pid_file(pid_file_path);
+	if (!pid_file)
+		throw Except("Failed to open PID file");
+	pid_file << getpid();
+	pid_file.close();
+}
+ 
+int 
+open_device()
+{
+	int fd;
+
+	// Try to open the device
+	while ((fd = open(dev_path, O_RDWR)) == -1)
+	{
+		if (stop_flag.load(std::memory_order_acquire)) {
+			std::cout << "Stop flag is set, exiting device open loop" << std::endl;
+			return -1;
+		}
+ 
+		if (errno != ENOENT) {
+			std::cerr << "Critical error opening device" << std::endl;
+			throw Except("Can't open device");
 		}
 
-		for (i = 0; i < cpu_cnt; ++i) {
-			std::packaged_task<void(int, int, std::string, std::string,
-						std::string)> task(run_thread);
-			futures.push_back(task.get_future());
-			threads.emplace_back(std::move(task), i, fd,
-					     vm["host"].as<std::string>(), user, password);
+		std::cout << "Device not found, will retry..." << std::endl;
+		std::this_thread::sleep_for(wait_for_dev);
+	}
+
+	std::cout << "Successfully opened device: " << dev_path << std::endl;
+	return fd;
+}
+
+void 
+run_worker_threads(int fd, size_t cpu_count)
+{
+	std::vector<std::thread> threads;
+	std::vector<std::future<void>> futures;
+
+	for (size_t i = 0; i < cpu_count; ++i)
+	{
+		std::packaged_task<void(int, int, TfwLoggerConfig)> task(run_thread);
+		futures.push_back(task.get_future());
+		threads.emplace_back(std::move(task), i, fd, config);
+	}
+
+	spdlog::info("All worker threads started");
+
+	// Wait for all threads to complete
+	for (auto& future : futures)
+		future.get();
+ 
+	// Clean up threads
+	for (auto &t : threads)
+		if (t.joinable())
+			t.join();
+}
+
+int main(int argc, char *argv[])
+{
+	int fd = -1, res = 0;
+	std::shared_ptr<spdlog::logger> logger = nullptr;
+	fs::path config_path = "/etc/tempesta/tfw_logger.json";
+
+	try
+	{		
+		if (parse_basic_options(argc, argv, config_path) != CommandResult::CONTINUE) {
+			return 0;
 		}
 
-		spdlog::info("Daemon started");
+		load_configuration(argc, argv, config_path);
+
+		logger = setup_logger(config);
+		spdlog::info("Starting tfw_logger...");
 
-		for (auto& future : futures)
-			future.get();
+		size_t cpu_cnt = determine_cpu_count(config);
+
+		daemonize_if_needed(config);
+		set_sig_handlers();
+		create_pid_file(config.get_pid_file());
 
-		spdlog::info("Daemon stopped");
+		// Open the device
+		fd = open_device();
```

Comment:
Probably the comment is redundant - basically the function name is just the same as the comment